packages/@mantine/spotlight/src/spotlight.store.ts

```diff
@@ -52,9 +52,37 @@ export function setListId(id: string, store: SpotlightStore) {
   store.updateState((state) => ({ ...state, listId: id }));
 }
 
+function findElementByQuerySelector<T extends HTMLElement>(
+  root: Document | Element | ShadowRoot,
+  selector: string
+): T | null {
+  // Directly try to find the element in the current root.
+  const element = root.querySelector(selector);
+  if (element) return element as T;
```

Comment:
```suggestion
  const element = root.querySelector<T>(selector);
  if (element) return element;
```

How about handling it like this without using type assertions?

Reply:
Extending the HTMLElement type makes more sense, yes. Thanks!